### ISSUE TITLE
`indent` `CallExpression.arguments.first` incorrect for object arguments

### ISSUE THREAD

**Issue by rubensayshi** [eslint/eslint#7616]
**Tell us about your environment**

* **ESLint Version:** v3.10.2
* **Node Version:** v6.3.0
* **npm Version:** v3.10.3

**Please show your full configuration:**

```
{
    "rules": {
        "indent": [
            "error", 
            4,
            {
                "CallExpression": {
                    "arguments": "first"
                }
            }
        ]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
function myfn(a, b, c) {

}

myfn(
    "a", 
    "b", 
    {
        "c1": "",
        "c2": "" 
    }
);

myfn(
                "a", 
                "b", 
                "c"
);

myfn(
    "a", 
        "b", // Expected indentation of 4 spaces but found 8
    {
        "c1": "",
        "c2": "" 
    }
);

myfn(
        "a", 
        "b", 
        {// Expected indentation of 4 spaces but found 8
            "c1": "", // Expected indentation of 8 spaces but found 12
            "c2": ""  // Expected indentation of 8 spaces but found 12
        }// Expected indentation of 4 spaces but found 8
);
```

**What did you expect to happen?**
```
  22:9   error  Expected indentation of 4 spaces but found 8   indent
```

**What actually happened? Please include the actual, raw output from ESLint.**
```
  22:9   error  Expected indentation of 4 spaces but found 8   indent
  32:9   error  Expected indentation of 4 spaces but found 8   indent
  33:13  error  Expected indentation of 8 spaces but found 12  indent
  34:13  error  Expected indentation of 8 spaces but found 12  indent
  35:9   error  Expected indentation of 4 spaces but found 8   indent
```

it seems like it's expecting an object declaration `{}` to always be indented on +1 from the function call regardless of setting `CallExpression.arguments=first`

**Comment by kaicataldo** [comment]
Hey there, can I ask you to explain what you expected to happen? You've left an error message, but it's the same error message as the first error in the "what actually happened section", and I want to be sure we understand the issue correctly.

Are you saying you expected the following to error (it doesn't currently, which I think is correct, given that `"CallExpression": {"arguments": "first" }` is off by default)?
```js
myfn(
    "a", 
        "b", // Expected indentation of 4 spaces but found 8
    {
        "c1": "",
        "c2": "" 
    }
);
```

Or were you expecting the following not to error?

```js
myfn(
        "a", 
        "b", 
        {// Expected indentation of 4 spaces but found 8
            "c1": "", // Expected indentation of 8 spaces but found 12
            "c2": ""  // Expected indentation of 8 spaces but found 12
        }// Expected indentation of 4 spaces but found 8
);
```

Thanks!

**Comment by rubensayshi** [comment]
bleh, I've been copy pasting stuff too much around and made a mistake, the report is supposed to have `CallExpression.arguments=first` set, updated the report now!

the problem is the fact that `{}` needs to be on 8 spaces instead of the 12 following `CallExpression.arguments=first`.

I put the one on line 22 in to demonstrate the expected behaviour for `CallExpression.arguments=first` as it's the only error that I'm actually expecting!

**Comment by not-an-aardvark** [comment]
Thanks, I was able to reproduce this issue. Simpler reproduction case:

```js
/* eslint indent: [2, 2, {"CallExpression": {"arguments": "first"}}] */
foo(
    bar,
    {
      baz: 1
    }
);
```

The above case should not report an error, but it reports the curly braces in the object.

```
4:5 - Expected indentation of 2 spaces but found 4. (indent)
5:7 - Expected indentation of 4 spaces but found 6. (indent)
6:5 - Expected indentation of 2 spaces but found 4. (indent)
```

**Comment by not-an-aardvark** [comment]
Actually, I think this is a duplicate of https://github.com/eslint/eslint/issues/7320.

**Comment by rubensayshi** [comment]
yes, this seem like a duplicate, closing
